torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -2123,24 +2063,21 @@ class AsyncGatherCUDAWork : public AsyncGatherWork {
 
   void run() override {
     // Synchronize with copy operations.
-    at::cuda::OptionalCUDAGuard device_guard;
```

Comment:
no need to switch devices here? was this only necessary for cuda API?

Reply:
I don't see any operation below that seems to depend on the current device, do you?